Update config.go

redstone2019 opened the pull request:
Changed 2060 to 2060S, Added 2070S, 2080S, 2080TI, TITAN-RTX, and TITAN-V SKUs to select regions, removed -test flag and references.

redstone2019 committed:
Update config.go

Changed 2060 to 2060S, Added 2070S, 2080S, 2080TI, TITAN-RTX, and TITAN-V SKUs to select regions, removed -test flag and references.

redstone2019 commented:
ill remove the rest of the -test references later

ianmarmour commented on internal/config/config.go:
```diff
@@ -86,10 +85,23 @@ type Config struct {
 				SKU: strPtr("5394902900"),
 			},
-			"3070": {},
+			"2070S": {
```
I would prefer we just kept to the 2060 for now as it's in stock pretty much globally adding too many flags right now would confuse users.

ianmarmour replied:
Also changing what was previously working 2060 to 2060S would confuse users, probably should avoid making any breaking changes to the existing flags/input when possible.

redstone2019 replied:
ok on the 2060 thing, but can we add the new models to a section near the bottom called older generation cards?

ianmarmour replied:
I mean I'm okay with adding older models I guess but I want to be super consistent about the naming convention for those models, I'm not sure what S indicates? 

Like 2060, 2070, 2080, 2080TI, (I don't really like adding Titan as it's a totally different naming convention)

It might also make sense adding like a -list option flag so you could do like -region=USA -list to list models or something.

redstone2019 replied:
But if we explicitly say what each model is titan won't be a problem. I second doing some kind of -list flag. And the S is Super the only models they carry now.

redstone2019 replied:
I don't know how to make a new flag other than what I just did. Please help @ianmarmour

ianmarmour replied:
So I hear you regarding the SUPER for S but I do wonder if this is a case where we should do what users want not what they say. If super is the only model for sale maybe we support super without the flag, Titan cards I guess I can give way on that. My biggest thought here is I want to write code that is simple that I don't have to support long term, I.E. when NVIDIA decides to call them GGGTTTTXXXX SUPER MEGA ULTRA I can still just put 4080 lol.

redstone2019 replied:
true

redstone2019 replied:
I changed the lineup to 2060, 2070, 2080, 2080TI, 3080, and 3090 only now

ianmarmour reviewed (changes requested):
Also we should just make the config change in here, there's basically 2 lines that need to be removed from the main.go file to update it. 

https://github.com/ianmarmour/nvidia-clerk/blob/master/main.go#L37

https://github.com/ianmarmour/nvidia-clerk/blob/master/main.go#L82

And then the README.MD updates.

redstone2019 committed:
Update main.go

removed -test flag references

redstone2019 committed:
Update README.md

removed and replaced -test flag references

redstone2019 committed:
Update config.go

replaced all instances of 2060S with 2060

redstone2019 committed:
Update config.go

missed some testsku references, fixing error 2 on windows64 build

redstone2019 committed:
Update config_test.go

removed references of unknown variable TestSKU

ianmarmour commented on internal/config/config.go:
```diff
@@ -152,14 +197,30 @@ var regionalConfig = map[string]RegionalConfig{
 		Locale:       "en_gb",
 		NvidiaLocale: "en-gb",
-		Currency:     "EUR",
+		Currency:     "CZK",
```
Have you actually tested these? Like browsed to the URLs and tried the add to checkout functionality locally to ensure they work? Before we make changes we need to validate these locally.

redstone2019 replied:
yes

redstone2019 replied:
https://api-prod.nvidia.com/direct-sales-shop/DR/products/en_gb/CZK/5394902800

redstone2019 committed:
Update main.go

added -list flag still working on it

redstone2019 committed:
Update config.go

removed s suffix from models to make it less confusing since NVIDIA doesn't carry non-super cards anymore

redstone2019 committed:
Update main.go

Fixing error 2 by removing bool.flag

redstone2019 committed:
Update main.go

removed more -list stuff

redstone2019 committed:
Update main.go

added -list flag to the best of my ability

redstone2019 committed:
Update main.go

added "var list string" line 24

redstone2019 committed:
Update main.go

changed return nil to continue and added } to fix errors

redstone2019 committed:
Update main.go

fixed syntax, hopefully

redstone2019 committed:
Update main.go

testing fix

redstone2019 committed:
Update main.go

trying more fixes

redstone2019 committed:
Update main.go

redstone2019 committed:
Update main.go

redstone2019 committed:
Update main.go

ianmarmour commented on internal/config/config.go:
```diff
@@ -377,24 +585,56 @@ var regionalConfig = map[string]RegionalConfig{
+			"2080TI": {
+				SKU: strPtr("5218614400"),
+			},
+			"TITAN-RTX": {
```
So if every region as the 2060/2070/2080/2080TI lets just keep it to those, TITAN-RTX/TITAN-V seem like there only in some regions and I think that's rather confusing for folks.

redstone2019 replied:
ok

redstone2019 replied:
can we leave it but not document it so if you change your mind we could use it?

redstone2019 committed:
Update main.go

trying case

redstone2019 committed:
Merge branch 'master' into redstone2019-patch-1

ianmarmour commented on internal/config/config.go:
```diff
@@ -98,7 +96,20 @@ var regionalConfig = map[string]RegionalConfig{
+			"TITAN-RTX": {
+				SKU: strPtr("5256301000"),
+			},
+			"3070": {
```
Remove this.

ianmarmour commented on internal/config/config.go:
```diff
@@ -115,8 +126,16 @@ var regionalConfig = map[string]RegionalConfig{
+			"2080TI": {
+				SKU: strPtr("5218987100"),
+			},
 			"3070": {
```
Remove this.

ianmarmour commented on internal/config/config.go:
```diff
@@ -134,7 +153,17 @@ var regionalConfig = map[string]RegionalConfig{
+			"TITAN-RTX": {
+				SKU: strPtr("5254456900"),
+			},
+			"3070": {
```
Remove this.

ianmarmour commented on internal/config/config.go:
```diff
@@ -151,7 +180,23 @@ var regionalConfig = map[string]RegionalConfig{
+			"TITAN-V": {
+				SKU: strPtr("5144941300"),
+			},
+			"3070": {
```
Remove this.

ianmarmour commented on internal/config/config.go:
```diff
@@ -161,14 +206,30 @@ var regionalConfig = map[string]RegionalConfig{
+			"TITAN-V": {
+				SKU: strPtr("5144941100"),
+			},
+			"3070": {
```
Remove this.

ianmarmour commented on internal/config/config.go:
```diff
@@ -202,7 +278,17 @@ var regionalConfig = map[string]RegionalConfig{
+			"TITAN-RTX": {
+				SKU: strPtr("5256301100"),
+			},
+			"3070": {
```
Remove this.

ianmarmour commented on internal/config/config.go:
```diff
@@ -219,7 +305,20 @@ var regionalConfig = map[string]RegionalConfig{
+			"TITAN-RTX": {
+				SKU: strPtr("5256301000"),
+			},
+			"3070": {
```
Remove this.

ianmarmour commented on internal/config/config.go:
```diff
@@ -236,7 +335,23 @@ var regionalConfig = map[string]RegionalConfig{
+			"TITAN-V": {
+				SKU: strPtr("5144751300"),
+			},
+			"3070": {
```
Remove this.

ianmarmour commented on internal/config/config.go:
```diff
@@ -247,14 +362,26 @@ var regionalConfig = map[string]RegionalConfig{
+			"TITAN-RTX": {
+				SKU: strPtr("5256301200"),
+			},
+			"3070": {
```
Remove this.

ianmarmour commented on internal/config/config.go:
```diff
@@ -271,7 +398,20 @@ var regionalConfig = map[string]RegionalConfig{
+			"TITAN-RTX": {
+				SKU: strPtr("5256301200"),
+			},
+			"3070": {
```
Remove this.

ianmarmour commented on internal/config/config.go:
```diff
@@ -288,7 +428,20 @@ var regionalConfig = map[string]RegionalConfig{
+			"TITAN-RTX": {
+				SKU: strPtr("5256302000"),
+			},
+			"3070": {
```
Remove this.

ianmarmour commented on internal/config/config.go:
```diff
@@ -305,15 +458,31 @@ var regionalConfig = map[string]RegionalConfig{
+			"TITAN-V": {
+				SKU: strPtr("5144941000"),
+			},
+			"3070": {
```
Remove this.

ianmarmour commented on internal/config/config.go:
```diff
@@ -322,7 +491,17 @@ var regionalConfig = map[string]RegionalConfig{
+			"2080TI": {
+				SKU: strPtr("5218987100"),
+			},
+			"3070": {
```
Remove this.

ianmarmour commented on internal/config/config.go:
```diff
@@ -339,7 +518,20 @@ var regionalConfig = map[string]RegionalConfig{
+			"TITAN-RTX": {
+				SKU: strPtr("5256301600"),
+			},
+			"3070": {
```
Remove this.

ianmarmour commented on internal/config/config.go:
```diff
@@ -369,7 +564,20 @@ var regionalConfig = map[string]RegionalConfig{
+			"TITAN-RTX": {
+				SKU: strPtr("5256301400"),
+			},
+			"3070": {
```
Remove this.

ianmarmour commented on internal/config/config.go:
```diff
@@ -386,24 +594,56 @@ var regionalConfig = map[string]RegionalConfig{
+			"TITAN-V": {
+				SKU: strPtr("5145366000"),
+			},
+			"3070": {
```
Remove this.

ianmarmour commented on internal/config/config.go:
```diff
@@ -386,24 +594,56 @@ var regionalConfig = map[string]RegionalConfig{
+			"TITAN-V": {
+				SKU: strPtr("5144941200"),
+			},
+			"3070": {
```
Remove this.

redstone2019 replied:
yep

redstone2019 committed:
Update config.go

removed empty and titan model tags

redstone2019 committed:
Update main.go
